client/rest/src/main/java/org/elasticsearch/client/RestClient.java

```diff
@@ -615,16 +615,16 @@ private void setHeaders(HttpRequest httpRequest, Collection<Header> requestHeade
      */
     private NodeTuple<Iterator<Node>> nextNode() throws IOException {
         NodeTuple<List<Node>> nodeTuple = this.nodeTuple;
-        List<Node> hosts = selectHosts(nodeTuple, blacklist, lastNodeIndex, nodeSelector);
+        Iterable<Node> hosts = selectNodes(nodeTuple, blacklist, lastNodeIndex, nodeSelector);
```

Comment:
ops, I also renamed this method while I was at it, and made it return Iterable rather than a list. Hopefully that's ok.

Reply:
Fine with me.